### PR TITLE
Remove yarn

### PR DESCRIPTION
**Pull request by lamartire**
### Context
Removes `yarn` with all the following artifacts

#### Checklist
- [x] [Sign Allure CLA][cla]
- [x] Provide unit tests

[cla]: https://cla-assistant.io/accept/allure-framework/allure2

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -25,24 +25,34 @@ jobs:
         uses: actions/setup-node@v2
         with:
           node-version: ${{ matrix.node-version }}
-          cache: 'yarn'
 
-      - name: yarn install
+      - name: Restore npm cache
+        uses: actions/cache@v2
+        with:
+          path: |
```
Comment (mxschmitt) [inline .github/workflows/build.yml:32]: I think it's discouraged to cache node_modules and cache .npm instead see here: https://github.com/actions/cache/blob/main/examples.md#node---npm

Comment (lamartire) [reply .github/workflows/build.yml]: Good catch! Thank you for the comment 👍

**Comment by lamartire** [comment]
Hey guys. Build is red because I haven't fixed eslint issues (I think, they have low priority), but build and test steps are green – that's enough to ensure about safe merge here.

**Review by vovsemenv — CHANGES_REQUESTED** [review]
Why we need to add eslint deps in every package??
I created demo how it works without this changes https://github.com/vovsemenv/allure-js/pull/1.
Everything works, deps look cleaner and build become green.

File: packages/allure-mocha/package.json [hunk 2]
```diff
@@ -29,10 +29,18 @@
     "@types/glob": "^7.1.4",
     "@types/mocha": "^9.0.0",
     "@types/node": "^16.7.8",
+    "@typescript-eslint/eslint-plugin": "^5.31.0",
+    "@typescript-eslint/parser": "^5.31.0",
+    "allure-commandline": "^2.18.1",
```
Comment (vovsemenv) [inline packages/allure-mocha/package.json:34]: We don't need `allure-commandline` devDependency in every package.
`npm run allure-report --workspace 'packages/allure-mocha'` works like a charm without it

Comment (lamartire) [reply packages/allure-mocha/package.json]: 👍

**Review by vovsemenv — APPROVED** [review]
Looks good

**Comment by lamartire** [comment]
Better to left yarn due the publishing workflow

**Pull request closed without merging**